offload/plugins-nextgen/common/include/PluginInterface.h

```diff
@@ -1251,6 +1254,9 @@ struct GenericDeviceTy : public DeviceAllocatorTy {
   /// Internal representation for OMPT device (initialize & finalize)
   std::atomic<bool> OmptInitialized;
 #endif
+
+  /// The total per-block native shared memory that a kernel may use.
+  uint32_t MaxBlockSharedMemSize = 0;
```

Comment:
nit: for cuda, I think this is an int (https://developer.download.nvidia.com/compute/DevZone/docs/html/C/doc/html/group__CUDA__DEVICE_g9c3e1414f0ad901d3278a4d6645fc266.html); for hsa, it's defined as a size_t: https://rocm.docs.amd.com/projects/HIP/en/docs-develop/doxygen/html/hsa__ext__amd_8h.html, so we could use size_t instead of uint32_t here, too